### DISCUSSION TITLE
how to use a specific native library

### DISCUSSION THREAD

**Discussion by LiFan0525** [xerial/sqlite-jdbc#803] [Q&A] [answered]
I want to use the customized sqlite. Before that, I downloaded the corresponding version of dll from the sqlite official website ，and configured it according to USAGE.md：
-Dorg.sqlite.lib.path=/path/to/folder
-Dorg.sqlite.lib.name=your-custom.dll
![image](https://user-images.githubusercontent.com/48920893/199688505-0f756148-7100-4fb7-994d-121faad034c7.png)

The following exception occurred:
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._ open_ utf8([BI)V

I extracted the local library of sqlite-jdbc for a cross reference test, which can confirm that the problem is in the dll

I don't know much about the C language. I want to know how to configure it correctly, and whether the sqlite downloaded from the official website needs special processing before it can be used

**Comment by gotson** [comment, marked as answer]
You can't use the off the shelf DLL, because it will lack the JNI part that.

What is it you are trying to achieve exactly?

**Reply by LiFan0525** [reply]
I want to add a custom fts5 tokenizer，But don't know how to add to sqlite-jdbc

**Reply by gotson** [reply]
see https://github.com/xerial/sqlite-jdbc/issues/449

**Reply by LiFan0525** [reply]
Thank you for your answer. I have seen this issue before, so I want to try other ways
